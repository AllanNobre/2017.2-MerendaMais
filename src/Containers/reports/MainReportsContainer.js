import { connect } from 'react-redux';
import MainReportsScreen from '../../screens/reports/MainReportsScreen';
import { isLoading, isNotLoading } from '../../actions/applicationActions';
import { asyncGetCurrentSchedule, asyncUpdateSchedule } from '../../actions/schedulingVisitActions';

const mapStateToProps = state => (
  {
    scheduleVisit: state.scheduleVisit,
    counselor: state.counselor,
    report: state.report,
    isLoading: state.application.isLoading,
    clickableView: state.application.clickableView,
  }
);

const mapDispatchToProps = dispatch => (
  {
    syncIsLoading: () => dispatch(isLoading()),
    syncIsNotLoading: () => dispatch(isNotLoading()),
    asyncGetCurrentSchedule: getData => dispatch(asyncGetCurrentSchedule(getData)),
    asyncUpdateSchedule: postData => dispatch(asyncUpdateSchedule(postData)),
  }
);

const MainReportsContainer = connect(mapStateToProps, mapDispatchToProps)(MainReportsScreen);

export default MainReportsContainer;
