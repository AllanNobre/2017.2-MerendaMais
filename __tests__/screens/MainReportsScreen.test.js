import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
// imported as a connected component!
import MainReportsContainer from '../../src/Containers/reports/MainReportsContainer';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  report: {
  },
  application: {
    isLoading: false,
  },
};

const store = mockStore(initialState);

describe('Testing MainReports Screen', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <MainReportsContainer />,
      { context: { store } },
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
