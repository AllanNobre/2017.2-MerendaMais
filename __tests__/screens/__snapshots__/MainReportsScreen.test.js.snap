// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DocCheckout Screen renders as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <Header
    backButton={true}
    subTitle=""
    title="Relatórios"
  />
  <ScrollViewMock>
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            Cozinha
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            Refeitório
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            Documentação
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            Arredores da Escola
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            Estoque de Alimentos
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            Qualidade da Alimentação
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            Manipuladores de Alimentos
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            Abastecimento de Água e Esgoto
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            Preparação e Distribuição de Alimentos
          </Text>
        </TouchableOpacity>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingTop": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "blue",
                "fontSize": 20,
                "paddingLeft": 20,
                "paddingRight": 10,
                "paddingTop": 5,
              }
            }
          >
            + Outras informações
          </Text>
        </TouchableOpacity>
      </View>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "backgroundColor": "#FF9500",
            "borderRadius": 7,
            "borderWidth": 1,
            "marginBottom": 20,
            "marginHorizontal": 15,
            "marginTop": 30,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "textAlign": "center",
            }
          }
        >
          Anexar fotos
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        style={
          Object {
            "backgroundColor": "#FF9500",
            "borderRadius": 7,
            "borderWidth": 1,
            "justifyContent": "flex-end",
            "marginBottom": 20,
            "marginHorizontal": 15,
            "marginTop": 30,
            "paddingVertical": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "textAlign": "center",
            }
          }
        >
          Gerar Relatório Final
        </Text>
      </TouchableOpacity>
    </View>
  </ScrollViewMock>
</View>
`;
